File: 3115-maximum-prime-difference/3115-maximum-prime-difference.cpp
class Solution {
public:
    int maximumPrimeDifference(vector<int>& nums) {
        auto m = false;
        auto n = false;
        auto i = 0;
        auto j = nums.size() - 1;

        while (i <= j) {

            if(!m)
                if(!IsPrime(nums[i])) i++;
                else m = true; 

            if(!n)
                if(!IsPrime(nums[j])) j--;
                else n = true;
            if(m && n) break; 
        }
        return j - i;
    }

private:
    bool IsPrime(int n) {
        if (n == 1) return false;
        for (auto i = 2; i <= n / 2; i++)
            if (n % i == 0) return false;
        return true;
    }
};

File: 3115-maximum-prime-difference/3115-maximum-prime-difference.java
class Solution {

    static boolean isPrime(int n){
        if(n==1) return false;
        if(n==2) return true;
        for(int i=2; i*i<=n; i++){
            if(n%i==0) return false;
        }
        return true;
    }
    public int maximumPrimeDifference(int[] arr) {

        int a = -1;
        int b = -1;
        int n = arr.length;
        for(int i=0; i<n; i++){
            if(isPrime(arr[i])){
                a = i;
                break;
            }
        }

        for(int i=n-1; i>=0; i--){
            if(isPrime(arr[i])){
                b = i;
                break;
            }
        }
        return b-a;
    }
}